python/hidet/runtime/storage.py

```diff
@@ -54,9 +54,11 @@ def memory_info(self) -> (int, int):
 
 class CudaMemoryAPI(MemoryAPI):
     def malloc(self, nbytes: int) -> int:
+        if nbytes == 0:
+            return 0
         with hidet.cuda.device(self.device.id):
-            addr = hidet.cuda.malloc_async(nbytes)
-        if addr == 0 and nbytes != 0:
+            addr = hidet.cuda.malloc(nbytes)
+        if addr == 0:
```

Comment:
When we call `cudaMalloc` with `nbytes==0`, it will return 0. Thus, we also need to check `nbytes != 0` to make sure it is indeed out-of-memory. The `nbytes==0` will happen when we use tensor like `hidet.randn(shape=[0])`.

Reply:
I see, then it might be the feature of cudaMallocAsync.

I did not see the new added check for nbytes==0. Then it looks good for me.